packages/@angular/cli/commands/lint.ts

```diff
@@ -45,17 +45,26 @@ export default Command.extend({
       `
     }
   ],
-  run: function (commandOptions: LintCommandOptions) {
+  run: function (commandOptions: LintCommandOptions, args: string[]) {
     const LintTask = require('../tasks/lint').default;
 
     const lintTask = new LintTask({
       ui: this.ui,
       project: this.project
     });
 
+    const configs = CliConfig.fromProject().config.lint || [];
+
+    if (args.length) {
+      configs.forEach(config => {
+        config.files = args;
+        config.project = '';
```

Comment:
This disables any rule that requires type information.  Unfortunately, the underlying issue being worked around here of specifying files that are not part of the program is a blocker for this feature.

I think some further thought on the semantics of passing files on the command line and the interaction with the defined configurations is needed.  As well as a more general assessment of the desired behavior when files are specified either in the configuration or command line that are not present within the program (behavior could be different).  TSLint has a pending PR that issues an explicitly worded fatal error when this condition occurs.  My suggestion would be an additional option that filters out the errant files.  Thoughts?

Reply:
That's a very good point. Previously I was not aware of tslint using type information for evaluating of some rules, thanks! 

Another way to implement this which I was considering is to pass files array deeper where the files are being evaluated and just use them to filter what was retrieved. So if I pass file `foo.ts` as args... It will only be used to remove everything which is not `foo.ts` but it will not add itself. This should prevent issue when typescript compilation fails as with the issue described in tslint repo.

What are your thoughts?

Reply:
How about only filtering the output by the files names that we want to have information ?

Like ng lint *services* ->

Do linting to all the files and report only errors on the files in the filter ?